### PR TITLE
get all roles then

### LINKED ISSUE #61: Get Roles
on the "get roles" block, the "for each" option doesn't work

### PR DESCRIPTION
**Pull request by noxyyk**

### CODE REVIEW

**Review by Malix-Labs — DISMISSED** [review]
- [ ] ~~Add a documentation link about this fix~~

**Review by Malix-Labs — COMMENTED** [review]
- [ ] Add a documentation link about this fix

File: src/blocks/discord/roles/get_all_roles_then.js [hunk 1]
```diff
@@ -33,6 +33,6 @@ Blockly.Blocks[blockName] = {
 Blockly.JavaScript[blockName] = function (block) {
     const server = Blockly.JavaScript.valueToCode(block, "SERVER", Blockly.JavaScript.ORDER_ATOMIC);
     const statementThen = Blockly.JavaScript.statementToCode(block, "THEN");
-    let code = `${server}.roles.cache.forEach(async (ro) =>{ \n ${statementThen} \n})\n`;
+    let code = `${server}._roles.cache.forEach(async (ro) =>{ \n ${statementThen} \n})\n`;
     return code;
-};
\ No newline at end of file
+};
```
Comment (Malix-Off) [inline src/blocks/discord/roles/get_all_roles_then.js:38]: Nice EOF Newline Fixing

**Comment by noxyyk** [comment]
It was still broken due to the cache, I tried it without it and it works!

File: src/blocks/discord/roles/get_all_roles_then.js [hunk 2]
```diff
@@ -33,6 +33,6 @@ Blockly.Blocks[blockName] = {
 Blockly.JavaScript[blockName] = function (block) {
     const server = Blockly.JavaScript.valueToCode(block, "SERVER", Blockly.JavaScript.ORDER_ATOMIC);
     const statementThen = Blockly.JavaScript.statementToCode(block, "THEN");
-    let code = `${server}.roles.cache.forEach(async (ro) =>{ \n ${statementThen} \n})\n`;
+    let code = `${server}._roles.forEach(async (ro) =>{ \n ${statementThen} \n})\n`;
```
Comment (Malix-Labs) [inline src/blocks/discord/roles/get_all_roles_then.js:36]: Why is there a `_` in `_roles`?

Comment (RedMan13) [reply src/blocks/discord/roles/get_all_roles_then.js:36]: dont ask me ask the d.js v13 team

**Comment by Malix-Labs** [comment]
@scratch-for-discord/toolbox please review

**Review by RedMan13 — APPROVED** [review]
aproved

File: src/blocks/discord/roles/get_all_roles_then.js (hunk 2 above)
Comment (Malix-Labs) [reply src/blocks/discord/roles/get_all_roles_then.js:36]: correct answer

**Pull request merged**